service/history/replication/executable_task.go

```diff
@@ -640,6 +640,17 @@ func (e *ExecutableTaskImpl) SyncState(
 		TargetClusterId:     int32(targetClusterInfo.InitialFailoverVersion),
 	})
 	if err != nil {
+		logger := log.With(e.Logger,
+			tag.WorkflowNamespaceID(syncStateErr.NamespaceId),
+			tag.WorkflowID(syncStateErr.WorkflowId),
+			tag.WorkflowRunID(syncStateErr.RunId),
+			tag.ReplicationTask(e.replicationTask),
+		)
+		var unFlushedBufferedEventsErr *UnFlushedBufferedEventsError
+		if errors.As(err, &unFlushedBufferedEventsErr) {
```

Comment:
this works across network?

Reply:
I think so, because UnFlushedBufferedEventsError is a new error type I defined at this pr.